Atomic/backendutils.py

```diff
@@ -75,16 +75,18 @@ def get_backend_for_container(self, container, str_preferred_backend=None):
         # Check preferred backend first
         if str_preferred_backend:
             be = self.get_backend_from_string(str_preferred_backend)
-            if be.has_container(container):
-                return be
+            con_obj = be.has_container(container)
+            if con_obj:
+                return be, con_obj
             # Didnt find in preferred, need to remove it from the list now
             del backends[self._get_backend_index_from_string(str_preferred_backend)]
 
         container_in_backends = []
         for backend in backends:
             be = backend()
-            if be.has_container(container):
-                container_in_backends.append(be)
+            con_obj = be.has_container(container)
+            if con_obj:
+                container_in_backends.append((be, con_obj))
         if len(container_in_backends) == 1:
             return container_in_backends[0]
```

Comment:
No, it returns an instance of the backend and a container object seen there on line 89.